Gordon360/Services/HousingService.cs

```diff
@@ -531,4 +557,228 @@ public bool ChangeApplicationDateSubmitted(int applicationID)
+        }
+        return result.ToArray();
+    }
+
+    /// <summary>
+    /// Gets an array of preferences of this user
+    /// </summary>
+    /// <returns> An array of preferences </returns>
+    public IEnumerable<HousingPreferenceViewModel> GetUserPreference(string username)
+    {
+        var email = context.ACCOUNT.FirstOrDefault(a => a.AD_Username == username)?.email;
+        var applicationID = context.Applicant.FirstOrDefault(a => (a.Applicant1 == email) && (a.Active == 1))?.ApplicationID;
+        var preference = context.Preference.Where(a => a.ApplicationID == applicationID).Select(ph => (HousingPreferenceViewModel)ph);
+        return preference.ToArray();
+    }
+
+    /// <summary>
+    ///Gets an array of preferred halls
+    /// </summary>
+    /// <returns> AN array of preferred halls </returns>
+    public IEnumerable<HousingPreferredHallViewModel> GetAllPreferredHalls()
+    {
+        var activeApplicants = context.Applicant.Where(a => a.Active == 1);
+
+        var result = context.PreferredHall
+        .Join(activeApplicants, ph => ph.ApplicationID, a => a.ApplicationID, (ph, a) => new HousingPreferredHallViewModel
+        {
+            ApplicationID = ph.ApplicationID,
+            Rank = ph.Rank,
+            HallName = ph.HallName,
+        }).Distinct();
+        if (result == null || !result.Any())
+        {
+            throw new ResourceNotFoundException() { ExceptionMessage = "The preferred halls could not be found." };
+        }
+        return result.ToArray();
+    }
+
+    /// <summary>
+    /// Gets an array of preferred hall of the user
+    /// </summary>
+    /// <returns> An array of preferences </returns>
+    public IEnumerable<HousingPreferredHallViewModel> GetUserPreferredHall(string username)
+    {
+        var email = context.ACCOUNT.FirstOrDefault(a => a.AD_Username == username)?.email;
+        var applicationID = context.Applicant.FirstOrDefault(a => (a.Applicant1 == email) && (a.Active == 1))?.ApplicationID;
+        var preferredHall = context.PreferredHall.Where(a => a.ApplicationID == applicationID).Select(ph => (HousingPreferredHallViewModel)ph);
+        return preferredHall.ToArray();
+    }
+
+    /// <summary>
+    ///Gets an array of applicants
+    /// </summary>
+    /// <returns> AN array of applicants </returns>
+    public IEnumerable<HousingApplicantViewModel>  GetAllApplicants()
+    {
+        var result = context.Applicant.Where(a => a.Active == 1).Select(a => (HousingApplicantViewModel) a);
+        if (result == null || !result.Any())
+        {
+            throw new ResourceNotFoundException() { ExceptionMessage = "The applicants could not be found." };
+        }
+        return result.ToArray();
+    }
+
+    /// <summary>
+    /// Gets an array of preferences
+    /// </summary>
+    /// <returns> An array of preferences </returns>
+    public IEnumerable<HousingApplicantViewModel> GetUserRoommate(string username)
```

Comment:
```suggestion
    public IEnumerable<HousingApplicantViewModel> GetUserRoommates(string username)
```
Since this returns an enumerable, it should be plural.